src/borg/repository.py

```diff
@@ -650,18 +650,25 @@ def report_error(msg):
         try:
             transaction_id = self.get_transaction_id()
             current_index = self.open_index(transaction_id)
-        except Exception:
+            logger.debug('Read committed index of transaction %d', transaction_id)
+        except Exception as exc:
             transaction_id = self.io.get_segments_transaction_id()
             current_index = None
+            logger.debug('Failed to read committed index (%s), using segments transaction %s', exc, transaction_id)
```

Comment:
just to be clear:
transaction_id in 659 can be None, so it will say ... "using segments transaction None", which is obviously not true, it won't use that.